shared/src/publication/Locator.ts

```diff
@@ -0,0 +1,225 @@
+/* Copyright 2021 Readium Foundation. All rights reserved.
+ * Use of this source code is governed by a BSD-style license,
+ * available in the LICENSE file present in the Github repository of the project.
+ */
+
+import { arrayfromJSONorString, numberfromJSON } from '../util/JSONParse';
+
+/**
+ * One or more alternative expressions of the location.
+ * https://github.com/readium/architecture/tree/master/models/locators#the-location-object
+ */
+export class LocatorLocations {
+  /** Contains one or more fragment in the resource referenced by the `Locator`. */
+  public readonly fragments: Array<string>;
+
+  /** Progression in the resource expressed as a percentage (between 0 and 1). */
+  public readonly progression?: number;
+
+  /** Progression in the publication expressed as a percentage (between 0 and 1). */
+  public readonly totalProgression?: number;
+
+  /** An index in the publication (>= 1).*/
+  public readonly position?: number;
+
+  /** Additional locations for extensions. */
+  public readonly otherLocations?: Map<string, any>;
+
+  /**
+   * Creates a [Locations].
+   */
+  constructor(values: {
+    fragments?: Array<string>;
+    progression?: number;
+    totalProgression?: number;
+    position?: number;
+    otherLocations?: Map<string, any>;
+  }) {
+    this.fragments = values.fragments ? values.fragments : new Array<string>();
+    this.progression = values.progression;
+    this.totalProgression = values.totalProgression;
+    this.position = values.position;
+    this.otherLocations = values.otherLocations;
+  }
+
+  /**
+   * Parses a [Locations] from its RWPM JSON representation.
+   */
+  public static deserialize(json: any): LocatorLocations | undefined {
+    if (!json) return;
+    const progression = numberfromJSON(json.progression);
+    const totalProgression = numberfromJSON(json.totalProgression);
+    const position = numberfromJSON(json.position);
+
+    const otherLocations = new Map<string, any>();
+
+    const reservedKeys = new Set([
+      'fragment',
+      'fragments',
+      'progression',
+      'totalProgression',
+      'position',
+    ]);
+    Object.entries(json).forEach(([key, value]) => {
+      if (!reservedKeys.has(key)) {
+        otherLocations.set(key, value);
+      }
+    });
+
+    return new LocatorLocations({
+      fragments: arrayfromJSONorString(json.fragments || json.fragment),
+      progression:
+        progression !== undefined && progression >= 0 && progression <= 1
+          ? progression
+          : undefined,
+      totalProgression:
+        totalProgression !== undefined &&
+        totalProgression >= 0 &&
+        totalProgression <= 1
+          ? totalProgression
+          : undefined,
+      position: position !== undefined && position > 0 ? position : undefined,
```

Comment:
`undefined > 0` is `false`, so you can just put:

```suggestion
      position: position > 0 ? position : undefined,
```